src/main/java/org/spongepowered/common/item/merchant/SpongeVillagerRegistry.java

```diff
@@ -0,0 +1,353 @@
+import net.minecraft.village.MerchantRecipeList;
+import org.spongepowered.api.data.type.Career;
+import org.spongepowered.api.data.type.Profession;
+import org.spongepowered.api.item.merchant.TradeOffer;
+import org.spongepowered.api.item.merchant.VillagerRegistry;
+import org.spongepowered.api.item.merchant.generator.TraceableTradeOfferGenerator;
+import org.spongepowered.api.item.merchant.generator.TradeOfferGenerator;
+import org.spongepowered.common.entity.SpongeCareer;
+import org.spongepowered.common.entity.SpongeProfession;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Random;
+
+public class SpongeVillagerRegistry implements VillagerRegistry {
+
+    private static final SpongeVillagerRegistry instance = new SpongeVillagerRegistry();
+    private static final ITradeList[] EMPTY_TRADES = new ITradeList[0];
+
+    private SpongeVillagerRegistry() {
+    }
+
+    public static VillagerRegistry getInstance() {
+        return instance;
+    }
+
+    @Override
+    public Map<Integer, List<TradeOfferGenerator>> getTradeOffers(final Career career) {
+        checkNotNull(career, "career");
+        checkArgument(career instanceof SpongeCareer, "Unsupported Career: %s (%s)", career.getClass().getName());
+        final Profession profession = checkNotNull(career.getProfession(), "profession");
+        checkArgument(profession instanceof SpongeProfession, "Unsupported Profession: %s (%s)", profession.getName(), profession.getClass()
+                .getName());
+        final SpongeCareer spongeCareer = (SpongeCareer) career;
+        final SpongeProfession spongeProfession = (SpongeProfession) profession;
+        // The types starting with 1
+        final int careerType = spongeCareer.type;
+        final int professionType = spongeProfession.type;
+
+        final ITradeList[][][][] defaultTrades = EntityVillager.DEFAULT_TRADE_LIST_MAP;
+        if (defaultTrades == null || defaultTrades.length <= professionType) {
+            throw new IllegalStateException("Could not find valid entry in default trade offer generators");
+        }
+        final ITradeList[][][] professionTrades = defaultTrades[professionType];
+        if (professionTrades == null || professionTrades.length <= careerType) {
+            throw new IllegalStateException("Could not find valid entry in profession trade offer generators " + profession.getName());
+        }
+        final ITradeList[][] careerTrades = professionTrades[careerType - 1];
+        if (careerTrades == null) {
+            throw new IllegalStateException("Could not find valid entry in career trade offer generators " + career.getName());
+        }
+        final ImmutableMap.Builder<Integer, List<TradeOfferGenerator>> result = ImmutableMap.builder();
+        for (int level = 1; level < careerTrades.length + 1; level++) {
+            final ITradeList[] levelTrades = careerTrades[level - 1];
```

Comment:
This is ridiculous, there is literally no reason to shift the level in this way.

Should be
```java
for (int level = 0; level < careerTrades.length; level++) {
    final ITradeList[] levelTrades = careerTrades[level];
```